### PR TITLE
build(composer): streamline local development setup #2798

### LINKED ISSUE #2798: build(composer): streamline local development setup
Instead of having to run `npm install` separately, we can hook it to composer's `post-update-cmd` event. This can simplify the setting up of the development environment for contributors.

This will also help setting up git's pre-commit hooks that is currently under development since npm will install eslinter and the pre-commit hooks will lint JS files before committing. 

What will be required from contributors is that they already have the following installed:

- Git
- Composer
- NPM
- PHPCS

@kevinwhoffman @ravinderk let me know your thoughts on this.

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #2798 

## Description
Now, "npm install" will execute automatically after composer's install and update commands finish their execution.

## How Has This Been Tested?
Manually tested

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,10 +22,12 @@
...
   "scripts": {
     "post-install-cmd": [
-      "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs"
+      "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs",
+      "npm install"
     ],
     "post-update-cmd": [
-      "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs"
+      "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs",
+      "npm install"
```
Comment (kevinwhoffman) [inline composer.json]: Let's remove the second `npm install`. I don't think we want to trigger `npm install` every time a developer runs `composer update`. The goal should be to streamline the initial setup of Give in a local environment, so it makes sense to hook into `composer install` but not `composer update`.

File: composer.json [hunk 2]
```diff
@@ -22,10 +22,12 @@
   },
   "scripts": {
     "post-install-cmd": [
-      "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs"
+      "\"vendor/bin/phpcs\" --config-set installed_paths vendor/wp-coding-standards/wpcs",
+      "npm install"
```
Comment (kevinwhoffman) [inline composer.json]: `npm install` is good here 👍

File: composer.json (hunk 1 above)
Comment (Sidsector9) [reply composer.json]: @kevinwhoffman The first one `post-install-cmd` will work only with a lock file present, which won't be the case in the initial setup.

`post-update-cmd` is for both update or install _without_ the lock file present, which will be the scenario after cloning the repo.

Comment (kevinwhoffman) [reply composer.json]: @Sidsector9 A `composer.lock` file should be committed to the Git repo. I created an issue for it in https://github.com/WordImpress/Give/issues/2896. Please make sure `composer.lock` gets committed and then continue with my recommendation above to remove the second mention of `npm install`.

Comment (Sidsector9) [reply composer.json]: @kevinwhoffman , I'll do that :+1:

**Comment by kevinwhoffman** [comment]
@Sidsector9 Your `npm install` script was firing correctly but it did not take care of the initial build necessary to put Give in a usable state because SCSS and JS assets were not processed. So I added a new `start` script that looks likes this:

`"start": "npm install && npm run watch"`

The end result is that running `composer install` will take care of the entire build process all the way through to launching browsersync. I'm going to hold off on merging until @DevinWalker takes a look but I think we've got what we're looking for. Thanks for getting this up and running.

**Comment by Sidsector9** [comment]
@kevinwhoffman Sounds good! My idea was just to install all the dependencies in a single step. Although, I should have aimed to setup the project to its usable state. 

One suggestion, can we use `npm run dev` instead of `npm run watch`, if the aim is to put the environment in a usable state?

**Comment by kevinwhoffman** [comment]
@Sidsector9 `npm run watch` does the same thing as `npm run dev` but it has the added bonus of launching browsersync so that your local site appears in the browser right away.

**Pull request merged**